MultipleFileUpload/MultipleFileUpload.php

```diff
@@ -74,23 +78,22 @@ public static function init()
 		$uiReg->register("MultipleFileUpload\\UI\\Plupload");
 
 		// Set default check callback
-		self::$validateFileCallback = callback(__CLASS__, "validateFile");
+		self::$validateFileCallback = [__CLASS__, "validateFile"];
 
-		// TODO: remove this magic
-		self::$baseWWWRoot = Environment::getHttpRequest()->url->baseUrl . "MultipleFileUpload/";
+        self::$baseWWWRoot = self::$container->getByType('Nette\Http\IRequest')->url->baseUrl . "MultipleFileUpload/";
 	}
 
 
 	/**
 	 * Register MFU into Nette
 	 */
-	public static function register()
+	public static function register($container)
 	{
-		self::init();
-
-		$application = Environment::getApplication();
-		$application->onStartup[] = callback(__CLASS__, "handleUploads");
-		$application->onShutdown[] = callback(__CLASS__, "cleanCache");
+        self::$container = $container;
+        self::init();
+		$application = $container->getByType('Nette\Application\Application');
+		$application->onStartup[] = [__CLASS__, "handleUploads"];
+		$application->onShutdown[] = [__CLASS__, "cleanCache"];
```

Comment:
@akyn84 This whole registration should be moved into Container Extension? Are you going to handle that?